scripts/update-nsis-functions.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Author:  Enrico Tröger
+# License: GPL v2 or later
+#
+# Updates the `functions` and `variables` entries in data/filetypes.nsis.
+
+set -e
+
+TOKENS_CPP_FILE="/tmp/nsis_tokens.cpp"
+TOKENS_CPP_URL="https://raw.githubusercontent.com/kichik/nsis/master/Source/tokens.cpp"
+BUILD_CPP_FILE="/tmp/nsis_build.cpp"
+BUILD_CPP_URL="https://raw.githubusercontent.com/kichik/nsis/master/Source/build.cpp"
+DATA_FILE=data/filedefs/filetypes.nsis
+
+[ -f "${DATA_FILE}" ]
+
+# download tokens.cpp and build.cpp from NSIS
+wget --quiet --output-document="${TOKENS_CPP_FILE}" "${TOKENS_CPP_URL}"
+wget --quiet --output-document="${BUILD_CPP_FILE}" "${BUILD_CPP_URL}"
+
+normalize() {
+   sort | tr '[:upper:]' '[:lower:]' | tr '\n' ' ' | tr '\r' ' ' | sed 's/^[[:blank:]]*//;s/[[:blank:]]*$//'
```

Comment:
You could merge the `tr` calls like that: `tr '[:upper:][:space:]' '[:lower:] '`

Reply:
Done in https://github.com/geany/geany/pull/2181/commits/81c370e83b77fb4e45609569731dc1a5e6125033. And added a comment so future-me will remember what is happening :).